src/auth0-session/client/edge-client.ts

```diff
@@ -0,0 +1,241 @@
+    }
+    const response = await oauth.authorizationCodeGrantRequest(
+      as,
+      client,
+      parameters,
+      redirectUri,
+      checks.code_verifier as string,
+      {
+        additionalParameters: extras.exchangeBody,
+        ...(clientPrivateKey && { clientPrivateKey }),
+        ...this.httpOptions()
+      }
+    );
+
+    const result = await oauth.processAuthorizationCodeOpenIDResponse(
+      as,
+      client,
+      response,
+      checks.nonce,
+      checks.max_age
+    );
+    if (oauth.isOAuth2Error(result)) {
+      throw new IdentityProviderError({
+        message: result.error_description || /* c8 ignore next  */ result.error,
+        error: result.error,
+        error_description: result.error_description
+      });
+    }
+    return result;
+  }
+
+  async endSessionUrl(parameters: EndSessionParameters): Promise<string> {
+    const [as] = await this.getClient();
+    const issuerUrl = new URL(as.issuer);
+
+    if (
+      this.config.idpLogout &&
+      (this.config.auth0Logout || (issuerUrl.hostname.match('\\.auth0\\.com$') && this.config.auth0Logout !== false))
+    ) {
+      const { id_token_hint, post_logout_redirect_uri, ...extraParams } = parameters;
+      const auth0LogoutUrl: URL = new URL(urlJoin(as.issuer, '/v2/logout'));
+      post_logout_redirect_uri && auth0LogoutUrl.searchParams.set('returnTo', post_logout_redirect_uri);
+      auth0LogoutUrl.searchParams.set('client_id', this.config.clientID);
+      Object.entries(extraParams).forEach(([key, value]: [string, string]) => {
+        if (value === null || value === undefined) {
+          return;
+        }
+        auth0LogoutUrl.searchParams.set(key, value);
+      });
+      return auth0LogoutUrl.toString();
+    }
+    if (!as.end_session_endpoint) {
+      throw new Error('no rp inititated logout');
+    }
+    const oidcLogoutUrl = new URL(as.end_session_endpoint);
+    Object.entries(parameters).forEach(([key, value]: [string, string]) => {
+      if (value === null || value === undefined) {
+        return;
+      }
+      oidcLogoutUrl.searchParams.set(key, value);
+    });
+
+    oidcLogoutUrl.searchParams.set('client_id', this.config.clientID);
+    return oidcLogoutUrl.toString();
+  }
+
+  async userinfo(accessToken: string): Promise<Record<string, unknown>> {
+    const [as, client] = await this.getClient();
+    const response = await oauth.userInfoRequest(as, client, accessToken, this.httpOptions());
+
+    return oauth.processUserInfoResponse(as, client, oauth.skipSubjectCheck, response);
+  }
+
+  async refresh(refreshToken: string, extras: { exchangeBody: Record<string, any> }): Promise<TokenEndpointResponse> {
+    const [as, client] = await this.getClient();
+    const res = await oauth.refreshTokenGrantRequest(as, client, refreshToken, {
+      additionalParameters: extras.exchangeBody,
+      ...this.httpOptions()
+    });
+    const result = await oauth.processRefreshTokenResponse(as, client, res);
```

Comment:
Same as above.

Reply:
This doesn't throw, if there's an error `oauth.isOAuth2Error(result)` is `true` (which is caught below)